revocation/crl/cache/bundle.go

```diff
@@ -0,0 +1,199 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"archive/tar"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+)
+
+const (
+	// BaseCRL is the file name of the base CRL
+	PathBaseCRL = "base.crl"
+
+	// Metadata is the file name of the metadata
+	PathMetadata = "metadata.json"
+)
+
+// Bundle is in memory representation of the Bundle tarball, including base CRL
+// file and metadata file, which may be cached in the file system or other
+// storage
+//
+// TODO: consider adding DeltaCRL field in the future
+type Bundle struct {
+	BaseCRL  *x509.RevocationList
+	Metadata Metadata
+}
+
+// Metadata stores the metadata infomation of the CRL
+//
+// TODO: consider adding DeltaCRL field in the future
+type Metadata struct {
+	BaseCRL  FileInfo  `json:"base.crl"`
+	CreateAt time.Time `json:"createAt"`
+}
+
+// FileInfo stores the URL and creation time of the file
+type FileInfo struct {
+	URL string `json:"url"`
+}
+
+// NewBundle creates a new CRL store with tarball format
+func NewBundle(baseCRL *x509.RevocationList, url string) (*Bundle, error) {
+	return &Bundle{
+		BaseCRL: baseCRL,
+		Metadata: Metadata{
+			BaseCRL: FileInfo{
+				URL: url,
+			},
+			CreateAt: time.Now(),
+		},
+	}, nil
+}
+
+// ParseBundleFromTarball parses the CRL blob from a tarball
+//
+// The tarball should contain two files:
+// - base.crl: the base CRL in DER format
+// - metadata.json: the metadata of the CRL
+//
+// example of metadata.json:
+//
+//	{
+//	  "base.crl": {
+//	    "url": "https://example.com/base.crl"
+//	  },
+//	  "createAt": "2024-07-20T00:00:00Z"
+//	}
+func ParseBundleFromTarball(data io.Reader) (*Bundle, error) {
```

Comment:
nit: code style: name convention
```suggestion
func ParseBundleFromTar(data io.Reader) (*Bundle, error) {
```

Reply:
Updated.